system-tests/test.py

```diff
@@ -219,6 +219,12 @@ def set_up(self):
         if self.test_parameters.work_directory.is_dir():
             shutil.rmtree(self.test_parameters.work_directory)
 
+    def tear_down(self):
+        """
+        Clean up config files following the run.
+        """
+        os.remove(self.conf_file)
```

Comment:
Since `self.conf_file` is a `Path` object I think you could use one of its methods here?  `self.conf_file.unlink()` maybe?